### ISSUE TITLE
Unable to start docker container after updating to latest

### ISSUE THREAD

**Issue by exterrestris** [alexta69/metube#121]
After updating to the latest docker container metube will not start, and errors out with the following:

```
metube_1   | Traceback (most recent call last):
metube_1   |   File "app/main.py", line 71, in <module>
metube_1   |     dqueue = DownloadQueue(config, Notifier())
metube_1   |   File "/app/app/ytdl.py", line 187, in __init__
metube_1   |     self.queue = PersistentQueue(self.config.STATE_DIR + '/queue')
metube_1   |   File "/app/app/ytdl.py", line 142, in __init__
metube_1   |     with shelve.open(path, 'c'):
metube_1   |   File "/usr/local/lib/python3.8/shelve.py", line 243, in open
metube_1   |     return DbfilenameShelf(filename, flag, protocol, writeback)
metube_1   |   File "/usr/local/lib/python3.8/shelve.py", line 227, in __init__
metube_1   |     Shelf.__init__(self, dbm.open(filename, flag), protocol, writeback)
metube_1   |   File "/usr/local/lib/python3.8/dbm/__init__.py", line 95, in open
metube_1   |     return mod.open(file, flag, mode)
metube_1   | _gdbm.error: [Errno 19] No such device: '/downloads/.metube/queue'
```

Reverting to the docker image from 4 weeks ago works fine

**Comment by exterrestris** [comment]
Perhaps maintaining tags of previous versions on Docker Hub would be a good idea, as reverting is non-trivial at present (and impossible if you no longer have the previous image locally)

**Comment by alexta69** [comment]
There's probably some sort of permission or access problem with your downloads folder. It can't create the .metube directory there, or the file within it. Where is the /downloads folder mounted? When you download a video, is it able to write the video file there?

Regarding keeping old tags, it's a good idea. I'll add it.

**Comment by PikuZheng** [comment]
Yes, in many docker environments, using the suggested `user 1001:1001` can cause permission issues. In my practice, I'll set `./favicon ./app` with permission 755 and `../.cache` with permission 777. 
But in your issue, I think you need to check the `/downloads` folder mapping in the container configuration.

**Comment by exterrestris** [comment]
I doubt there are any permissions issues - I run the Metube container as it's own user `metube`, which owns the downloads folder mapped into the container. I did try deleting the `./metube` folder and letting Metube recreate it, but that didn't help.

Relevant section of my docker-compose.yml:
```
  metube:
    image: alexta69/metube
    user: "206:1004"
    expose:
      - "8081"
    volumes:
      - "/mnt/storage/downloads/metube:/downloads"
    networks:
      - "proxy"
    environment:
      - "YTDL_OPTIONS={\"cookiefile\":\"/downloads/cookies.txt\"}"
    restart: unless-stopped
```

UID 206 is the `metube` user. GID 1004 is a `downloads` group which `metube` belongs to. `/mnt/storage/downloads/metube` is owned by `metube` and the `downloads` group, and has 775 permissions

`/mnt/storage` is a mergerfs mount

**Comment by alexta69** [comment]
Your configuration sounds right, and it sounds like you know what you're doing, so I really can't say what the problem is. Since I can't reproduce it on my side, it will take some debugging on your part to find it. I would suggest trying to map /tmp instead of /mnt/storage/downloads/metube to the /downloads location, just to see if it makes any difference. Also, you could execute a shell in the container (`docker exec -ti metube sh`), and in that shell try to write to the same location where MeTube tries and fails.

**Comment by exterrestris** [comment]
OK, after a quick check through the commit history to see what's changed in the last month, I've updated my docker-compose.yml to explicitly set the `STATE_DIR`, and have mounted it to a different directory (where I keep config etc. for my other docker containers). I've then pulled the latest image, and it works perfectly. Therefore, I suspect the problem is something to do with having the state directory on a mergerfs mount as opposed to directly on a disk.

As I've managed to work around the problem, feel free to close this issue
